### ISSUE TITLE
use_build_context_synchronously false positive in StatefulWidget

### ISSUE THREAD

**Issue by slavap** [dart-lang/linter#3465]
Method of StatefulWidget gives false warning:

```dart
Future<String> _getStr(BuildContext context) async {
  var a = await aaa();
  return mounted ? bbb(context, a) : '';
}
```
No warning if changed like that:
```dart
Future<String> _getStr(BuildContext context) async {
  var a = await aaa();
  if (mounted) {
    return bbb(context, a);
  } else {
    return '';
  }
}
```

**Comment by srawlins** [comment]
Another one for you, @goderbauer 😄

**Comment by goderbauer** [comment]
Looks like it's a false positive. The lint may not be considering the trinary operator properly?

**Comment by pq** [comment]
> The lint may not be considering the trinary operator properly?

Seems likely.

Thanks!

**Comment by spixy** [comment]
false warning also with:
```dart
final result = mounted && await MyDialog.show<bool>(context);
```
